### DISCUSSION TITLE
p-treeSelect should support leafOnly selection

### DISCUSSION THREAD

**Discussion by NateRadebaugh** [primefaces/community#848] [PrimeNG] [answered]
I want to use the treeselect component to provide an organized drop-down but only allow the leaf nodes to be selectable.

This means things like:

- Don't show check boxes for non-leaf nodes
- clicking on the non-leaf node should expand/collapse that section

This doesn't appear to be possible at the moment

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

Unfortunately, this is not possible at the moment

**Reply by NateRadebaugh** [reply]
Yes thank you, I'm proposing a possible improvement to this component to make it possible.

There's a number of inconsistencies between the inputs and function of `p-multiSelect` and `p-treeSelect` that make it difficult to easily swap between the two.

In the meantime, I will try to hack around it and if I come up with an alternative implementation I will provide here or propose a PR if the team is open to it.

**Comment by RobersonLuo** [comment]
I need to only select the leaf node too, but only can disable selecting the parent node is ok.
